src/Widgets/DataFormUploadWidget/DataFormUploadWidgetComponent.tsx

```diff
@@ -28,13 +28,25 @@ provideComponent(DataFormUploadWidget, ({ widget }) => {
   const onDropRejected = useCallback(() => setIsTooLarge(true), [])
 
   const { getRootProps, getInputProps, inputRef, isDragActive } = useDropzone({
-      maxSize: MAX_FILE_SIZE,
-      onDropAccepted,
-      onDropRejected,
+    maxSize: MAX_FILE_SIZE,
+    onDropAccepted,
+    onDropRejected,
     onDrop: (acceptedFiles) => {
-      setFiles(acceptedFiles)
+      setFiles((prevFiles) => {
+        const newFiles = acceptedFiles.filter((newFile) => {
+          return !prevFiles.some(
+            (existingFile) =>
+              existingFile.lastModified === newFile.lastModified &&
+              existingFile.name === newFile.name &&
+              existingFile.size === newFile.size &&
```

Comment:
* [ ] FYI Is there a strong argument to disallow dropping the same file multiple times? Garbage in garbage out? Actually adding two files by picking the same image once more was the first thing I tried. This may bite people in demos. Remove filtering?